src/transformers/generation/candidate_generator.py

```diff
@@ -398,15 +398,10 @@ def _crop_past_key_values(model, past_key_values, max_length):
         past_key_values.crop(max_length)
     elif past_key_values is not None:
         for idx in range(len(past_key_values)):
-            if past_key_values[idx] != ([], []):
-                new_past.append(
-                    (
-                        past_key_values[idx][0][:, :, :max_length, :],
-                        past_key_values[idx][1][:, :, :max_length, :],
-                    )
-                )
+            if past_key_values[idx] != ([],) * len(past_key_values[idx]):
```

Comment:
Mmm speedwise some stuff might break compile but not sure candidate generator was supported. Also sorry again for this break, introduced in Mllama 😢 will be in patch.